test/py/libvfio_user.py

```diff
@@ -595,18 +659,20 @@ def disconnect_client(ctx, sock):
     # notice client closed connection
     ret = vfu_run_ctx(ctx)
     assert ret == -1
-    assert c.get_errno() == errno.ENOTCONN
+    assert c.get_errno() == errno.ENOTCONN, os.strerror(c.get_errno())
 
 
 def get_reply(sock, expect=0):
     buf = sock.recv(4096)
     (msg_id, cmd, msg_size, flags, errno) = struct.unpack("HHIII", buf[0:16])
     assert (flags & VFIO_USER_F_TYPE_REPLY) != 0
-    assert errno == expect
+    assert errno == expect, "expect=%s, errno=%s" % (expect, errno)
```

Comment:
why? output already gives the values

Reply:
It doesn't for me:
```
    def test_dma_region_too_big():
        global ctx, sock

        payload = vfio_user_dma_map(argsz=len(vfio_user_dma_map()),
            flags=(VFIO_USER_F_DMA_REGION_READ |
                   VFIO_USER_F_DMA_REGION_WRITE),
            offset=0, addr=0x10000, size=MAX_DMA_SIZE + 4096)

>       msg(ctx, sock, VFIO_USER_DMA_MAP, payload, expect=123)

test/py/test_dma_map.py:65:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ test/py/libvfio_user.py:694: in msg
    return get_reply(sock, expect=expect)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sock = <socket.socket fd=4, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0, raddr=/tmp/vfio-user.sock.111900>, expect = 123

    def get_reply(sock, expect=0):
        buf = sock.recv(4096)
        (msg_id, cmd, msg_size, flags, errno) = struct.unpack("HHIII", buf[0:16])
        assert (flags & VFIO_USER_F_TYPE_REPLY) != 0
>       assert errno == expect
E       AssertionError

test/py/libvfio_user.py:671: AssertionError
```

Reply:
what's your version of pytest-3?

```
[john.levon@john-levon master]$ pytest-3 --version
This is pytest version 2.9.2, imported from /usr/lib/python3.6/site-packages/pytest.py
[john.levon@john-levon master]$ pytest-3  -rP --quiet test/py/test_dma_unmap.py 
...F.....
...

sock = <socket.socket fd=10, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0, raddr=/tmp/vfio-user.sock.126582>, expect = 22

    def get_reply(sock, expect=0):
        buf = sock.recv(4096)
        (msg_id, cmd, msg_size, flags, errno) = struct.unpack("HHIII", buf[0:16])
        assert (flags & VFIO_USER_F_TYPE_REPLY) != 0
>       assert errno == expect
E       assert 95 == 22

test/py/libvfio_user.py:605: AssertionError
```

Reply:
```
# pytest-3 --version
This is pytest version 3.10.1, imported from /usr/lib/python3/dist-packages/pytest.py
```

Reply:
I'd ask upstream somewhere, but I don't think we need code changes.